electrode-ota-server-model-acquisition/src/acquisition.js

```diff
@@ -35,6 +35,8 @@ export default (options, dao, weighted, _download, manifest, logger) => {
                     }
                 }
 
+                pkg = await dao.getNewestApplicablePackage(params.deploymentKey, params.tags);
```

Comment:
Doesn't getNewesetApplicablePackage() just replace deploymentForKey() at 26?  We call deployemntForKey() and then discard the result.  We don't need to make the call? 
Also, pkg returned from getNewestApplicablePackage() can be undefined.

Reply:
deploymentForKey() fetches the deployment details.  The value it resolves to in the `deployment` variable is getting used further down, so i think that call is still needed.

Reply:
ah yes, your right.